Colors/Pallet/Pallet.cs

```diff
@@ -6,79 +6,107 @@
 
 namespace PalletLib
 {
+    public struct ColorAbundance
+    {
+        public Color PalletColor { get; set; }
+        public int Abundance { get; set; }
+
+        public ColorAbundance(Color color, int abundance)
+        {
+            PalletColor = color;
+            Abundance = abundance;
+        }
+    }
+
     public class Pallet
     {
         readonly Bitmap _bmp;
 
-        public Pallet(string fileName)
+        IList<ColorAbundance> _colors = new List<ColorAbundance>();
+
+        int this[Color color]
         {
-            if (File.Exists(fileName))
+            get => _colors.Where(c => c.PalletColor == color).Select(c => c.Abundance).FirstOrDefault();
+            set 
             {
-                try
+                if (_colors.Any(c => c.PalletColor == color))
                 {
-                    _bmp = new Bitmap(fileName);
+                    var colorAbundance = _colors.Where(c => c.PalletColor == color).FirstOrDefault();
+                    _colors.Remove(colorAbundance);
                 }
-                catch
+                if (value > 0)
                 {
-                    throw new Exception("Not a bitmap image!!!");
+                    _colors.Add(new ColorAbundance(color, value));
                 }
-            }
-            else
-            {
-                throw new FileNotFoundException($"File {fileName} does not exist!!!");
-            }
+            } 
         }
 
-        public IEnumerable<KeyValuePair<Color, int>> GetMostFrequentColors(int number)
+        public Pallet(Bitmap bmp)
         {
-            var colors = GetAllColors();
-            return colors.OrderByDescending(c => c.Value).Take(number);
+            _bmp = bmp;
+            //InitColors();
+            InitColorsForBiggerPicturesBruteForce();
         }
 
-        IDictionary<Color, int> GetAllColors()
+        public Pallet(Bitmap bmp, int downgradeSquares)
         {
-            var colors = new Dictionary<Color, int>();
+            _bmp = bmp;
+            InitColorsWithDownGrade(downgradeSquares);
+        }
 
+        public IEnumerable<ColorAbundance> GetMostCommonColors(int number)
+        {
+            return _colors.OrderByDescending(c => c.Abundance).Take(number);
+        }
+
+        void InitColorsForSmallPicturesOnlyBruteForce() // Looks nice, but awfully slow for bigger pictures
```

Comment:
you have the same logic in "ImageProcessor: public static int CountImageColors(string path)" - try to reuse it